build fail looks real
```
======================================================================
FAIL: test_default_to_phone_date_on_same_day (casexml.apps.case.tests.test_rebuild.TestActionSortKey)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py", line 536, in test_default_to_phone_date_on_same_day
    self._test(expected_actions)
  File "corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py", line 574, in _test
    self.assertEqual(action, sorted_actions[index])
AssertionError: None: update - 2001-01-01 00:00:01 (2001-01-01 00:00:00) != None: update - 2001-01-01 00:00:00 (2001-01-01 00:00:01)
```

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py
@@ -576,7 +576,10 @@ class TestActionSortKey(SimpleTestCase):
 EMPTY_DATE = object()
 
 
-def _make_action(server_date, phone_date=EMPTY_DATE, action_type=const.CASE_ACTION_UPDATE, user_id='someuserid'):
+def _make_action(server_date, phone_date=EMPTY_DATE, action_type=const.CASE_ACTION_UPDATE, user_id='someuserid', form_id=None):
     if phone_date == EMPTY_DATE:
         phone_date = server_date
-    return CommCareCaseAction(action_type=action_type, server_date=server_date, date=phone_date, user_id=user_id)
+    form_id = form_id or uuid.uuid4().hex
+    return CommCareCaseAction(
+        action_type=action_type, server_date=server_date, date=phone_date, user_id=user_id, xform_id=form_id
+    )

--- a/corehq/form_processor/backends/couch/update_strategy.py
+++ b/corehq/form_processor/backends/couch/update_strategy.py
@@ -394,7 +394,7 @@ def _action_sort_key_function(case):
                     raise MissingServerDate()
 
                 form_cmp = lambda form_id: (form_ids.index(form_id)
-                                            if form_id in form_ids else sys.maxint, form_id)
+                                            if form_id in form_ids else sys.maxint)
                 # if the user is the same you should compare with the special logic below
                 # if the user is not the same you should compare just using received_on
                 return (

--- a/corehq/form_processor/backends/couch/update_strategy.py
+++ b/corehq/form_processor/backends/couch/update_strategy.py
@@ -382,7 +382,7 @@ def _action_sort_key_function(case):
         else:
             form_ids = list(case.xform_ids)
 
-            if first_action.xform_id == second_action.xform_id:
+            if first_action.xform_id and first_action.xform_id == second_action.xform_id:
                 # short circuit if they are from the same form
                 return cmp(
                     _type_sort(first_action.action_type),

